marc.py

```diff
@@ -0,0 +1,612 @@
+        # Since this depends on the pool, it might be better not to cache it.
+        # But it's probably not a huge problem if it's outdated.
+        # File formats: a=one format, m=multiple formats, u=unknown
+        if len(pool.delivery_mechanisms) == 1:
+            file_formats_code = "a"
+        else:
+            file_formats_code = "m"
+        record.add_field(
+            Field(tag="007", data="cr cn ---" + file_formats_code + "nuuu"))
+
+        # Field 008 (fixed-length data elements):
+        data = datetime.datetime.now().strftime("%y%m%d")
+        publication_date = edition.issued or edition.published
+        if publication_date:
+            date_type = "s" # single known date
+            # Not using strftime because some years are pre-1900.
+            date_value = "%04i" % publication_date.year
+        else:
+            date_type = "n" # dates unknown
+            date_value = "    "
+        data += date_type + date_value
+        data += "    "
+        # TODO: Start tracking place of publication when available. Since we don't have
+        # this yet, assume everything was published in the US.
+        data += "xxu"
+        data += "                 "
+        language = "eng"
+        if edition.language and len(edition.language) == 3:
+            language = edition.language
+        data += language
+        data += "  "
+        record.add_field(
+            Field(tag="008", data=data))
+
+    @classmethod
+    def add_marc_organization_code(cls, record, marc_org):
+        record.add_field(
+            Field(tag="003", data=marc_org))
+
+    @classmethod
+    def add_isbn(cls, record, identifier):
+        # Add the ISBN if we have one.
+        isbn = None
+        if identifier.type == Identifier.ISBN:
+            isbn = identifier
+        if not isbn:
+            _db = Session.object_session(identifier)
+            for identifier_id in identifier.equivalent_identifier_ids()[identifier.id]:
+                equivalent = get_one(_db, Identifier, id=identifier_id)
+                if equivalent.type == Identifier.ISBN:
+                    isbn = equivalent
+                    break
+        if isbn:
+            record.add_field(
+                Field(
+                    tag="020",
+                    indicators=[" "," "],
+                    subfields=[
+                        "a", isbn.identifier,
+                    ]))
+
+    @classmethod
+    def add_title(cls, record, edition):
+        # Non-filing characters are used to indicate when the beginning of a title
+        # should not be used in sorting. This code tries to identify them by comparing
+        # the title and the sort_title.
+        non_filing_characters = 0
+        if edition.title != edition.sort_title and ("," in edition.sort_title):
+            sort_title = edition.sort_title
+            last_comma = len(sort_title) - sort_title[::-1].index(",")
+            stemmed = edition.sort_title[:last_comma-1]
```

Comment:
I think these three lines could be replaced with `stemmed = edition.sort_title[:edition.sort_title.rindex(",")]`